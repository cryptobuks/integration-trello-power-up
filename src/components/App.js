import React from 'react'
import GRAY_ICON from '../svg/invision-gray-16px.svg'
import { isInVisionUrl } from '../util/url.js'
import config from '../util/config.js'
import T from '../util/trello.js'

export default class App extends React.Component {
  constructor (props) {
    super(props)
    this.grayIcon = `${config.baseUrl}${GRAY_ICON}`

    this.cardSectionInit = this.cardSectionInit.bind(this)
    this.cardButtonInit = this.cardButtonInit.bind(this)
    this.cardButtonCallback = this.cardButtonCallback.bind(this)

    T.initialize({
      'attachment-sections': this.cardSectionInit,
      'card-buttons': this.cardButtonInit
    })
  }

  cardButtonCallback (t) {
    return t.popup({
      title: 'Attach prototype',
      url: './attach',
      height: 205
    })
  }

  cardSectionInit (t, options) {
    const claimed = options.entries.filter(attachment =>
      isInVisionUrl(attachment.url)
    )
    if (!claimed || claimed.length === 0) {
      return []
    }
    return [
      {
        id: 'InVision',
        claimed: claimed,
        icon: this.grayIcon,
        title: 'InVision',
        content: {
          type: 'iframe',
          url: t.signUrl('./section', {}),
          height: 50
        }
      }
    ]
  }

  cardButtonInit (t, options) {
    return [
      {
        icon: this.grayIcon,
        text: 'InVision',
        callback: this.cardButtonCallback
      }
    ]
  }
  render () {
    return (
      <div>
        InVision Trello Power-Up<br />
        Build date: {config.buildDate}
        <br />
        Version: {config.version}
      </div>
    )
  }
}
