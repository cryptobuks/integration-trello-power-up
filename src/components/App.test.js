import React from 'react'
import App from './App'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import T from '../util/trello.js'

jest.mock('../util/trello.js')
jest.mock('../util/config.js')
jest.mock('../util/auth.js')
jest.mock('../util/cover.js')

function setup () {
  const ctx = {}
  ctx.wrapper = shallow(<App />)
  ctx.component = ctx.wrapper.instance()
  expect(ctx.component).toBeDefined()
  expect(toJson(ctx.wrapper)).toMatchSnapshot()
  return ctx
}

describe('App', () => {
  it('renders renders and initializes TrelloPowerUp', () => {
    const ctx = setup()
    expect(T.initialize).toHaveBeenCalledWith({
      'attachment-sections': ctx.component.cardSectionInit,
      'card-buttons': ctx.component.cardButtonInit
    })
  })

  it('process button callback', () => {
    const ctx = setup()
    const t = {
      popup: jest.fn()
    }

    ctx.component.cardButtonCallback(t)
    expect(t.popup).toHaveBeenCalled()
  })

  it('should initialize card button', () => {
    const ctx = setup()
    var res = ctx.component.cardButtonInit()
    expect(res.length).toBe(1)
    expect(res).toContainEqual({
      icon: ctx.component.grayIcon,
      text: 'InVision',
      callback: ctx.component.cardButtonCallback
    })
  })

  it('should initialize card section with attachments', () => {
    const ctx = setup()
    var res = ctx.component.cardSectionInit(
      {
        signUrl: a => a
      },
      {
        entries: [
          { url: 'https://invis.io/BZCY6PFS3' },
          { url: 'https://google.com' }
        ]
      }
    )
    expect(res.length).toBe(1)
    expect(res).toContainEqual({
      id: 'InVision',
      claimed: [{ url: 'https://invis.io/BZCY6PFS3' }],
      icon: ctx.component.grayIcon,
      title: 'InVision',
      content: {
        type: 'iframe',
        url: './section',
        height: 50
      }
    })
  })

  it('should not innitialize card section with attachments', () => {
    const ctx = setup()
    var res = ctx.component.cardSectionInit(undefined, {
      entries: [{ url: 'https://google.com' }]
    })
    expect(res.length).toBe(0)
  })
})
