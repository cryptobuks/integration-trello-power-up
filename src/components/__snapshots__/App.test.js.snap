// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App process button callback 1`] = `
<div>
  InVision T
</div>
`;

exports[`App renders renders and initializes TrelloPowerUp 1`] = `
<div>
  InVision T
</div>
`;

exports[`App should initialize card button 1`] = `
<div>
  InVision T
</div>
`;

exports[`App should initialize card section with attachments 1`] = `
<div>
  InVision T
</div>
`;

exports[`App should not innitialize card section with attachments 1`] = `
<div>
  InVision T
</div>
`;
